thunder/executors/torch_autograd.py

```diff
@@ -165,6 +165,9 @@ def split_forward_backward(computation_trc: TraceCtx, compile_data, compile_stat
 
     # Now we can run the optimization passes on the forward trace
     # TODO Restore request for no rematerialization
+
+    # rematerialization could change the names, the new names used in rematerialization need to be new
+    fw_trace.names.update(bw_trace.names)
```

Comment:
If it's needed this should probably be done inside the rematerialization function, not here.

Reply:
this is here to pass all the already used names to the rematerialize function(inside `fusion_pass`, currently remat gets the already used names from its `trace` argument)